pydantic/typing.py

```diff
@@ -152,9 +159,10 @@ def is_callable_type(type_: AnyType) -> bool:
 
 
 if sys.version_info >= (3, 7):
+    literal_types = {type_ for type_ in (TypingLiteral, TypingExtensionsLiteral) if type_ is not None}
 
     def is_literal_type(type_: AnyType) -> bool:
-        return Literal is not None and getattr(type_, '__origin__', None) is Literal
+        return Literal is not None and getattr(type_, '__origin__', None) in literal_types
```

Comment:
In the case where the set has a single element (i.e., python 3.7), I benchmarked the `in` check vs the old `is` check at ~0.4s vs 0.5s slower over 10,000,000 iterations. This absolute magnitude of the time difference seems small enough to be worth the simplification, rather than adding another branch for `sys.version_info >= (3, 8)`.

But I could refactor to do that check instead if preferable.

Reply:
I also confirmed having `literal_types` be a set was ~20% faster than having it be a tuple, even with only two elements.

Reply:
Also, I'd rather not *require* the use of `typing.Literal` instead of `typing_extensions.Literal` when using python 3.8, since that will make it more annoying to migrate to python 3.8.

Reply:
`typing_extensions` imports `Literal` from `typing` in 3.8 so you can keep the old identity check and the effect will be the same.

Reply:
> `typing_extensions` imports `Literal` from `typing` in 3.8 so you can keep the old identity check and the effect will be the same.

great, that should make all these checks simpler.